python/ray/tune/suggest/hebo.py

```diff
@@ -0,0 +1,359 @@
+            you want to run first to help the algorithm make better suggestions
+            for future parameters. Needs to be a list of dicts containing the
+            configurations.
+        evaluated_rewards (list): If you have previously evaluated the
+            parameters passed in as points_to_evaluate you can avoid
+            re-running those trials by passing in the reward attributes
+            as a list so the optimiser can be told the results without
+            needing to re-compute the trial. Must be the same length as
+            points_to_evaluate. (See tune/examples/hebo_example.py)
+        random_state_seed (int, None): seed for reproducible
+            results. Defaults to None. Please note that setting this to a value
+            will change global random states for `numpy` and `torch`
+            on initalization and loading from checkpoint.
+
+    Tune automatically converts search spaces to HEBO's format:
+
+    .. code-block:: python
+
+        from ray import tune
+        from ray.tune.suggest.hebo import HEBOSearch
+
+        config = {
+            "width": tune.uniform(0, 20),
+            "height": tune.uniform(-100, 100)
+        }
+
+        hebo = HEBOSearch(metric="mean_loss", mode="min")
+        tune.run(my_func, config=config, search_alg=hebo)
+
+    Alternatively, you can pass a HEBO `DesignSpace` object manually to the
+    Searcher:
+
+    .. code-block:: python
+
+        from ray import tune
+        from ray.tune.suggest.hebo import HEBOSearch
+        from bo.design_space.design_space import DesignSpace
+
+        space_config = [
+            {'name' : 'width', 'type' : 'num', 'lb' : 0, 'ub' : 20},
+            {'name' : 'height', 'type' : 'num', 'lb' : -100, 'ub' : 100},
+        ]
+        space = DesignSpace().parse(space_config)
+
+        hebo = HEBOSearch(space, metric="mean_loss", mode="min")
+        tune.run(my_func, search_alg=hebo)
+
+    """
+
+    def __init__(
+            self,
+            space: Optional[Union[
+                Dict, "hebo.design_space.design_space.DesignSpace"]] = None,
+            metric: Optional[str] = None,
+            mode: Optional[str] = None,
+            points_to_evaluate: Optional[List[Dict]] = None,
+            evaluated_rewards: Optional[List] = None,
+            random_state_seed: Optional[int] = None,
+            **kwargs):
+        assert hebo is not None, (
+            "HEBO must be installed!. You can install HEBO with"
+            " the command: `pip install "
+            "git+https://github.com/huawei-noah/noah-research.git"
+            "#subdirectory=HEBO`.")
+        if mode:
+            assert mode in ["min", "max"], "`mode` must be 'min' or 'max'."
+        if random_state_seed:
+            assert random_state_seed is None or isinstance(
+                random_state_seed, int
+            ), "random_state_seed must be None or int, got '{}'.".format(
+                type(random_state_seed))
```

Comment:
```suggestion
        if random_state_seed is not None:
            assert isinstance(
                random_state_seed, int
            ), "random_state_seed must be None or int, got '{}'.".format(
                type(random_state_seed))
```

(otherwise we wouldn't check this for `random_state_seed=0.0`, as it evaluates to `False`)